src/hooks/use-element-width.ts

```diff
@@ -0,0 +1,11 @@
+import { useState } from "preact/hooks";
+import type { Ref } from "preact/hooks";
+import useResizeListener from "./resize-listener";
+
+function useElementWidth(ref: Ref<HTMLElement>) {
+  const [width, setWidth] = useState(undefined as number | undefined);
+  useResizeListener(() => setWidth(ref.current?.offsetWidth));
+  return width;
+}
+
+export default useElementWidth;
```

Comment:
is there a way to combine this and `useResizeListener` into one hook? or does that not make sense with "the react/preact way"™?

Reply:
Yeah, we could. I think having them separate may make sense if they were understandably composable / differentiable and we may plan to use each independently, but in this instance I think we really only care about width. Part of the reason I didn't flatten is because there's a bit of legacy to refactor. I'll see about doing that up-front or as a quick follow-on.

Reply:
> is there a way to combine this and `useResizeListener` into one hook? or does that not make sense with "the react/preact way"™?

Flattened in d8aab7d.